src/hotspot/share/runtime/thread.cpp

```diff
@@ -2496,20 +2496,9 @@ void JavaThread::verify_not_published() {
 // directly and when thread state is _thread_in_native_trans
 void JavaThread::check_safepoint_and_suspend_for_native_trans(JavaThread *thread) {
   assert(thread->thread_state() == _thread_in_native_trans, "wrong state");
-
   assert(!thread->has_last_Java_frame() || thread->frame_anchor()->walkable(), "Unwalkable stack in native->vm transition");
 
-  if (thread->is_external_suspend()) {
-    thread->java_suspend_self_with_safepoint_check();
```

Comment:
I found the equivalent of this part of the if-statement in
JavaThread::handle_special_runtime_exit_condition().